src/main/java/com/mraof/minestuck/world/ProspitDimension.java

```diff
@@ -0,0 +1,128 @@
+package com.mraof.minestuck.world;
+
+import com.mraof.minestuck.player.IdentifierHandler;
+import com.mraof.minestuck.skaianet.SburbConnection;
+import com.mraof.minestuck.skaianet.SkaianetHandler;
+import com.mraof.minestuck.util.Debug;
+import com.mraof.minestuck.world.biome.MSBiomes;
+import com.mraof.minestuck.world.gen.MSWorldGenTypes;
+import com.mraof.minestuck.world.gen.ProspitGenSettings;
+import com.mraof.minestuck.world.gen.SkaiaGenSettings;
+import com.mraof.minestuck.world.lands.LandTypePair;
+import net.minecraft.block.Blocks;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+import net.minecraft.world.biome.provider.BiomeProviderType;
+import net.minecraft.world.dimension.Dimension;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraft.world.gen.ChunkGenerator;
+import net.minecraftforge.common.ModDimension;
+
+import javax.annotation.Nullable;
+import java.util.function.BiFunction;
+
+public class ProspitDimension extends Dimension
```

Comment:
While the prospit and derse dimensions would have more different things than the lunar transportalizers, enough to warrant different classes, they do still share quite a bit of functionality, so I would suggest also adding an abstract class for the common stuff between them

Reply:
Sounds reasonable, I do intend for them to have different sky renders, but very similar terrain generation